platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/RosterUtils.java

```diff
@@ -338,4 +341,42 @@ public static AddressBook buildAddressBook(@NonNull final Roster roster) {
 
         return addressBook;
     }
+
+    /**
+     * Maps a {@link Network} object to its equivalent {@link Roster}.
+     *
+     * @param network the network to represent as a roster
+     * @return the converted roster
+     */
+    public static Roster fromNetwork(@NonNull final Network network) {
+        return fromMetadata(network.nodeMetadata());
+    }
+
+    /**
+     * Maps a list of {@link NodeMetadata} objects to the corresponding {@link Roster}.
+     *
+     * @param metadata the list of node metadata to represent as a roster
+     * @return the converted roster
+     */
+    public static Roster fromMetadata(@NonNull final List<NodeMetadata> metadata) {
+        return new Roster(metadata.stream()
+                .filter(Objects::nonNull)
+                .map(RosterUtils::entryFromMetadata)
+                .toList());
+    }
+
+    private static RosterEntry entryFromMetadata(@NonNull final NodeMetadata metadata) {
+        if (metadata.hasRosterEntry()) {
+            return metadata.rosterEntryOrThrow();
+        }
+
+        final var node = metadata.nodeOrThrow();
+        return RosterRetriever.buildRosterEntry(
+                NodeId.of(node.nodeId()),
+                node.weight(),
+                node.gossipCaCertificate(),
+                // If we're reading metadata, the endpoint order SHOULD match the address book, which always has the
+                // internal endpoint at index 0
+                node.serviceEndpoint());
```

Comment:
Then you cannot put it into the roster w/o flipping. A RosterEntry should always have the external endpoint at index 0.

Reply:
Agreed, this needs to be flipped.

Reply:
This only proves my earlier point: passing around two service endpoints in a list, with no designation of internal vs. external, is _at best_ difficult to account for. I recommend we change these APIs from an arbitrary list of `ServiceEndpoint` to something more explicit.

Reply:
I don't disagree with the sentiment fully. However, as of today, the plan is to use the endpoints in the RosterEntry to represent various external endpoints, such as IP-address based, FQDN-based, and in the future, perhaps IPv6-based. While currently it's unclear how this array of endpoints is going to be actually used, it is certain that we don't have plans to include any internal endpoints in this list because they're generally not useful in the roster as a public data structure.

But as long as we do use this list, we need to have the external endpoint at index zero, so the quoted code needs to flip the endpoints here.